### PR TITLE
fix(deps): bump zspec to v0.9.2 to init std.testing.io_instance (closes #583)

### LINKED ISSUE #583: build: `zig build test` exits 1 despite 456/456 tests passing
## Symptom

On \`feat/unify-scenes-prefabs\` (commit 280ef79), \`zig build test\` reports:

\`\`\`
Build Summary: 94/96 steps succeeded (1 failed); 456/456 tests passed
test transitive failure
+- run test failure
error: process exited with error code 1
failed command: ./.zig-cache/o/b486870aaf8eec6ac64054e697e26a22/test
\`\`\`

Every test passes (456/456), but a build step still fails and the command exits non-zero. Likely a stale install step, a missing artifact, or one test binary that links cleanly but exits non-zero before any test body runs.

## Reproduce

Inside a fresh Zig 0.16.0 environment with no \`.zig-cache\`:

\`\`\`bash
git clone https://github.com/labelle-toolkit/labelle-engine.git
cd labelle-engine
git checkout feat/unify-scenes-prefabs  # commit 280ef79 or later
zig build test
echo "exit: $?"
\`\`\`

Repro confirmed in a Docker container (Ubuntu 24.04 + Zig 0.16.0) on 2026-05-25. The same failure appeared across multiple runs.

## Impact

Low — tests are green, so the change is safe to merge. But CI gates that key on the exit code will mark the build red, and developers will second-guess whether \"failure\" means a real regression. Worth diagnosing before #582 lands.

## Suggested diagnosis steps

1. \`zig build test -j1\` to serialize and identify which specific binary exits non-zero
2. Inspect that binary's run output — likely it's a fixture, install step, or a test that runs a subprocess that's missing in the container's PATH
3. If it's an environment thing (missing dep), guard it with a runtime check rather than letting it fail silently

### PR DESCRIPTION
**Pull request by apotema**
## Summary

The CI deadlock under `registry_scan_spec` on x86_64-linux had **nothing to do with `Io.Threaded` worker-pool exhaustion or `prefab_cache.scanDir`**. The engine pinned `zspec` at v0.9.1, the release *before* [apotema/zspec#45](https://github.com/apotema/zspec/pull/45) ("init `std.testing.io_instance`"). v0.9.1's spec runner never ran `std.testing.io_instance = .init(allocator, .{})`, so the stdlib global stayed as Zig's undefined-pattern bytes (`0xaaaaaaaa…`).

The first `std.testing.io.*` call any spec made — concretely `tmpDir()`'s opening `io.random(...)` — then deadlocked deterministically:

1. `random(userdata = &io_instance, …)` casts the uninitialized bytes to `*Threaded` and hands them to `randomMainThread`.
2. `randomMainThread` calls `mutexLock(&t.mutex)`.
3. `t.mutex.state.raw == 0xaaaaaaaa` — neither `.unlocked` (0) nor `.locked_once` (1) nor `.contended` (2). The opening `cmpxchgStrong(.unlocked, .locked_once, …)` fails; the `swap(.contended, .acquire)` returns the garbage value, which compares `!= .unlocked`, so the thread enters `Thread.futexWaitUncancelable` on a futex no other thread will ever wake.

Reproduced under `docker --platform=linux/amd64 --cpus=2 -m 7g ubuntu:24.04`: a single TID parked in `futex_wait_queue` forever (confirmed via `/proc/$pid/task/*/wchan`). macOS and Windows masked the bug because their memory-init paths happened to leave the mutex bytes in a state `mutexLock` could recover from.

This PR bumps the `zspec` dependency hash to v0.9.2 — the first release that includes apotema/zspec#45's `io_instance` init. With that change the runner zero-initializes the global before any test runs, the mutex is `.unlocked` like it should be, and `tmpDir()` returns immediately.

Drops the Linux gate that #585 introduced as an emergency workaround. Re-exports `RegistryScanSpec` unconditionally so every platform now runs the full RFC #561 / #577 registry-scan coverage. Also corrects the `io_helper.io()` comment, which had attributed the original hang to dual-pool `sigaction` racing — the real cause is what's described above.

## Test plan

- [x] `docker --platform=linux/amd64 --cpus=2 -m 7g`: `zig build spec` passes 28/28 in **102 ms** (was: hang → exit 124).
- [x] Docker amd64 `zig build test` full suite green in **1 m 22 s** — well within the 10-min CI timeout #585 added.
- [x] macOS arm64 `zig build spec` 28/28 in **46 ms**; `zig build test` full suite green in **16 s**.
- [ ] CI run on this PR confirms the same on real `ubuntu-latest`.

### CODE REVIEW

**Comment by apotema** [comment]
@copilot review

File: build.zig.zon [hunk 1]
```diff
@@ -20,8 +20,17 @@
...
+            // `std.testing.io_instance` in the runner (apotema/zspec#45).
+            // The previously-pinned v0.9.1 left it as `undefined`, so the
+            // first `std.testing.io.*` call in a spec — e.g. `tmpDir()`'s
+            // inner `io.random()` — deadlocked on `mutexLock` because the
+            // mutex's state was the `0xaaaaaaaa` undefined-memory pattern
+            // instead of `.unlocked`. macOS/Windows happened to mask the
+            // bug (different memory-init paths); the x86_64-linux runner
+            // hit it deterministically. Closes #583.
+            .url = "https://github.com/apotema/zspec/archive/v0.9.2.tar.gz",
+            .hash = "zspec-0.9.1-jaKLbXgMBACFwbNjflhbMyP113leoYjboxn-1UOP-FGw",
```
[the remark this answers is not in the document]
Comment (apotema) [reply build.zig.zon:33]: Verified against the actual fetch — this is a wrong-direction correction. Running `zig fetch https://github.com/apotema/zspec/archive/v0.9.2.tar.gz` returns:

```
zspec-0.9.1-jaKLbXgMBACFwbNjflhbMyP113leoYjboxn-1UOP-FGw
```

The hash prefix is derived from the upstream package's own `build.zig.zon` `.version` field, and apotema/zspec v0.9.2 was tagged without bumping that internal version string (still `"0.9.1"`). The current hash matches `zig fetch` output verbatim; changing the prefix to `zspec-0.9.2-` would actually break dependency resolution. Declining.
